### PR TITLE
Add Bubblesort feature

### PR DESCRIPTION
**Pull request by fernandorpm**
# Add Bubblesort feature

## To add
- bubble_sort.rb
- bubble-sort.gif
- .rubocop.yml

## To update
- README.md
- MIT.md

### CODE REVIEW

**Review by cyonii — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️✍️

Good work @team, you have shown a great effort implementing this algorithm.
Your repository is neat 👏 

## To Improve
- [x] Your `bubble_sort_by` method fails to sort an array when a block is passed to it, please fix this issue.
- [x] For some reason unknown to me the Github action has refused to run, I suggest you try to fix this by pushing the `.github` directory to your `main` branch.
- [x] Your PR description should focus on the changes you intend to merge in, while your README should do the job of explaining what the program does.
- [x] Your README does not include how to set up this project on a local machine, please include that.

Kindly implement the required changes and resubmit for review.
Happy coding, and stay motivated

> #### You can find me on slack 👉 [CY Kalu](https://microverse-students.slack.com/team/U01494U96FR) if you need further assistance or clarification. I'm always available and will be glad to help.

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+def bubble_sort(arr)
+  has_changed = true
+
+  while has_changed == true
```
Comment (cyonii) [inline bubble_sort.rb]: Your `has_changed` variable is already boolean, there's no need testing for `has_changed == true`, the code below would work as fine
```
while has_changed
    # your code here
end
```

**Review by cyonii — CHANGES_REQUESTED** [review]

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      has_changed = true
+    end
+  end
+
+  bubble_sort(arr) if has_changed
+
+  arr
+end
+
+bubble_sort([14, 56, 2, 78, 9])
```
Comment (rukundoeric) [inline bubble_sort.rb:16]: Please remove this debugging  code

File: bubble_sort.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      has_changed = true
+    end
+  end
+
+  bubble_sort_by(arr) { |a, b| a.length - b.length } if has_changed
+
+  arr
+end
+
+bubble_sort_by(%w[ohayou hi hello hey horray]) { |a, b| a.length - b.length }
```
Comment (rukundoeric) [inline bubble_sort.rb]: Please remove this debugging  code

File: bubble_sort.rb (hunk 2 above)
Comment (rukundoeric) [inline bubble_sort.rb]: Please remove this debugging  code

**Review by rukundoeric — CHANGES_REQUESTED** [review]
# Status: ♻️ CHANGES REQUIRED ♻️
_Great job guys so far 👏 💯 👍. However, there are few issues we need to address before we move on, please check the comments below._

- ### To Highlight
  - [x] The project is clean. 💯
  - [x] The project follows Git flow. 💯

- ### To Improve
  - [x] [MANDATORY] Kindly remove all debugging codes and then request another code review.


_Please make all the required changes before you submit another code review. If you have any problems or concerns, you can reach out to me on Slack: Rukundo Eric 🚀 . I wish you all the very best 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/), 
Slack: Rukundo Eric**

**Review by jubaan — APPROVED** [review]
[![Microverse][microverse-shield]][microverse-url]
<div>
<h1 align="center"> 
  🎉🥳 🎊 🎉 🥳 🥳 🎊 🎉 🥳🥳 🎊 🎉 🥳 🎉 <br />
  🎉🥳 🎊 🎉 🥳 🥳 🎊 🎉 🥳🥳 🎊 🎉 🥳 🎉 <br />
  🥳 🎊 🎉 🥳  APPROVED 🥳 🎊 🎉 🥳 <br />
  🎉🥳 🎊 🎉 🥳 🥳 🎊 🎉 🥳🥳 🎊 🎉 🥳 🎉 <br />
  🎉🥳 🎊 🎉 🥳 🥳 🎊 🎉 🥳🥳 🎊 🎉 🥳 🎉
</h1>
<h2 align="center">
  What’s done is done. What’s gone is gone. <br />
  One of life’s lessons is always moving on. <br />
  It’s okay to look back to see how far you’ve come but <br /> 
  KEEP MOVING FORWARD.
</h2>

<details open="open">
  <summary>Review Summary</summary>
  <ol>
    <li><a href="#optional"> 💡Nice to have and other comments ✍🏼 ✍🏼 </a></li>
    <li><a href="#contact"> 🧔🏼 About your Reviewer </a> (you don't know it... but we are just automated bots 🤖) </li>
    <li><a href="#help"> 🆘 <b> 😭 Feeling stuck? or Need help</b>? 🙏🏼 🙏🏼 </a></li>
  </ol>
</details>
</div>


<h3 id="optional"> 1. 💡 Nice to have and other comments </h3>
<details close="close">
<summary> [click to open] If you have something here, even when approved, it's just a little push to improve your project.</h3></summary>


- [ ] Nothing more to add.


</details>

<h3 id="contact"> 2. 🧔🏼 About your Reviewer </h3>
<details close="close">
<summary> [click to open] Say "hello world!" on Slack: @jubaan</summary>
<br />
<div align="center">
  <a src="https://www.github.com/jubaan">
    Julio Añoveros
  </a>
  <p>
    Julio Añoveros | @jubaan</br></br>
    <b>Full-stack Web Developer</b> and</br>
    🤖 Code Reviewer at <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a>:rocket:
  </p>
</div>
</details>

<h3 id="help"> 3. 🆘 <b> 😭 Feeling stuck? or Need help</b>? 🙏🏼 🙏🏼</h3>
<p> How does it feel? Is nice to advance in the program, right? It keeps you motivated even with all the struggles and takes you one step closer to the goal 🎯!. I wan't you to keep advancing so if you feel stuck and/or need help, remember there is always someone you can reach to keep moving forward, doesn't mater how "dum" you think the questions is.</p>
<details close="close">
  <summary> [click to open] <b>Feel stuck?</b> </summary>
  <br>
  <p> First breath... just do it. Take 15-20 min away from the computer, stretch, meditate, do some exercise, drink water, or do whatever is necessary to distract your mind from the obstacle.</p>
  <p><b>Sometimes you just need to get away to see things with a new perspective.</b></p>
</details>

<details close="close">
  <summary> [click to open] <b>Need help?</b> </summary>
  <br>
  <p> Send me a DM in <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> Students Slack, you'll find me as @jubaan. If I'm not available there are other Code Reviewers you can contact. They'll help as well.</p>
  <p>Remember that <a href="https://www.microverse.org/?grsf=kfm2ej">Microverse</a> has other channels where you can get an answer to your questions:</p>
  <ol>
    <li>The <b>students channel 🧑‍🎓 </b> in slack,</li>
    <li>your <b>teammates 👥 </b>,</li>
    <li>your <b>mentor 👨🏼 👩🏼 </b>, and</li>
    <li><b>Student Success 🏫 </b>.</li>
  </ol>
  <p><b> DON'T hesitate to ask! </b></p>
</details>
<p>If all of the above doesn't help... get a <a href="https://rubberduckdebugging.com/">rubber duck</a> ... it would be great to own one 😄. </p>

<h1 align="center"> See you a the top 🚀!! </h1>

[microverse-shield]: https://img.shields.io/badge/-microverse-%236F23FF?style=for-the-badge
[microverse-url]: https://www.microverse.org/?grsf=kfm2ej
[bootstrap-shield]: https://img.shields.io/badge/Bootstrap-white?style=for-the-badge&logo=bootstrap
[bootstrap-url]: https://www.microverse.org/?grsf=kfm2ej

**Pull request merged**